sky/compositor/raster_cache.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SKY_COMPOSITOR_RASTER_CACHE_H_
+#define SKY_COMPOSITOR_RASTER_CACHE_H_
+
+#include <memory>
+#include <unordered_map>
+
+#include "base/macros.h"
+#include "third_party/skia/include/core/SkSize.h"
+#include "third_party/skia/include/core/SkImage.h"
+#include "sky/compositor/instrumentation.h"
+#include "sky/engine/wtf/PassRefPtr.h"
+#include "sky/engine/wtf/RefPtr.h"
+
+namespace sky {
+namespace compositor {
+
+class RasterCache {
+ public:
+  RasterCache();
+  ~RasterCache();
+
+  RefPtr<SkImage> GetImage(SkPicture* picture, const SkISize& physical_size);
+  void Sweep();
+
+ private:
+  struct Entry {
+    Entry();
+    ~Entry();
+
+    bool used = false;
+    bool reused_image = false;
```

Comment:
No, I was using it to gather stats, but I removed the stats code.  I can remove the flag if you like.